packages/apps/plugins/plugin-stack/src/StackPlugin.tsx

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { definePlugin, PluginDefinition } from '@dxos/react-surface';
+
+import { StackMain } from './components';
+import { isStack } from './props';
+import translations from './translations';
+
+export const StackPlugin: PluginDefinition = definePlugin({
+  meta: {
+    id: 'dxos:stack',
+  },
+  provides: {
+    translations,
+    component: (datum, role) => {
+      switch (role) {
+        case 'main':
+          if (Array.isArray(datum) && isStack(datum[0])) {
```

Comment:
why are we passed an array?

Reply:
my instinct says to avoid tuples when an object with two keys could be used, so as not to step on the space occupied by arity. i.e. its nice to be able to not worry about special tuples when dealing with arrays, to be able to assume they're arrays of a uniform type - this enables techniques a la graphql, etc. Switching to an object with two keys makes it much easier to duck type with `in` operators.